tsdb/head.go

```diff
@@ -987,6 +1004,76 @@ func (h *Head) truncateMemory(mint int64) (err error) {
 	return nil
 }
 
+func (h *Head) WaitForPendingReadersInTimeRange(mint, maxt int64) {
+	// TODO: should we have some limit on how long we wait in total? If so, how long?
```

Comment:
Not forever. It will be blocked till all the queriers that overlap with this truncation are closed that were got _before_ truncation started. So it depends on how long the queries run. Should be few mins max before they are timed out?

Reply:
Yeah, so that directly depends on the query timeout, which is set via the `--query.timeout` flag, 2m by default. (It can be set on a per-query basis via the HTTP API, but it is still capped by the value set via the flag.) If the query timeout is really limiting the wait time here implicitly, I don't think we need to introduce another separate timeout here.

Reply:
Meaning: Perhaps remove the TODO and change it into an explanation that the query timeout limits the wait time here implicitly.